cl/scrapers/tasks.py

```diff
@@ -439,30 +166,25 @@ def extract_recap_pdf(
             processed.append(pk)
             continue
 
-        with NamedTemporaryFile(
-            prefix="extract_file_",
-            suffix=".pdf",
-            buffering=0,  # Make sure it's on disk when we try to use it
-        ) as tmp:
-            tmp.write(rd.filepath_local.read())
-            process = make_pdftotext_process(tmp.name)
-            content, err = process.communicate()
-            content = content.decode()
-
-            if needs_ocr(content):
-                if not skip_ocr:
-                    # probably an image PDF. Send it to OCR.
-                    success, content = extract_by_ocr(tmp.name)
-                    if success:
-                        rd.ocr_status = RECAPDocument.OCR_COMPLETE
-                    elif content == "" or not success:
-                        content = "Unable to extract document content."
-                        rd.ocr_status = RECAPDocument.OCR_FAILED
-                else:
-                    content = ""
-                    rd.ocr_status = RECAPDocument.OCR_NEEDED
-            else:
-                rd.ocr_status = RECAPDocument.OCR_UNNECESSARY
+        response = microservice(
+            service="document-extract",
+            item=rd,
+            params={"ocr_available": not skip_ocr},
+        )
+        if not response.ok:
+            print("Error from microservice")
+            continue
+
+        content = response.json()["content"]
+        extracted_by_ocr = response.json()["extracted_by_ocr"]
+        if skip_ocr and content or content and not extracted_by_ocr:
+            rd.ocr_status = RECAPDocument.OCR_UNNECESSARY
+        elif extracted_by_ocr and content:
+            rd.ocr_status = RECAPDocument.OCR_COMPLETE
+        elif extracted_by_ocr and not content:
+            rd.ocr_status = RECAPDocument.OCR_FAILED
+        else:
+            rd.ocr_status = RECAPDocument.OCR_NEEDED
```

Comment:
I *think* this might be more clear? It uses Python 3.10's new [structural pattern matching](https://peps.python.org/pep-0636/). I like that it makes it more clear that we've covered all the bases (except one, spot it!). I think it makes it more clear which cases belong with which circumstances as well, as you're reading. For example, I read it as "if it has content and it was extracted by OCR, then...", "if it has content but wasn't extracted by OCR, check if if skipped OCR on purpose..." 

And so forth. There's probably a better way to make this more "structured" or something. What do you think? I also experimented with re-nesting the if statements, but that got kind of deeply nested, which prompted me to play with this.

```suggestion
        has_content = bool(content)
        extracted_by_ocr = response.json()["extracted_by_ocr"]
        match has_content, extracted_by_ocr:
            case True, True:
                rd.ocr_status = RECAPDocument.OCR_COMPLETE
            case True, False:
                if skip_ocr:
                    rd.ocr_status = RECAPDocument.OCR_UNNECESSARY
            case False, True:
                rd.ocr_status = RECAPDocument.OCR_FAILED
            case False, False:
                rd.ocr_status = RECAPDocument.OCR_NEEDED    
```

Reply:
oh boy... cases